google-cloud-speech/lib/google/cloud/speech/v1/helpers.rb

```diff
@@ -86,6 +87,124 @@ def streaming_recognize streaming_config, options: nil
+            proto_op = @operations_client.get_operation name, options: options
+            wrap_proto_op_with_gax_op proto_op, options
+          end
+
+          ##
+          # Lists operations that match the specified filter in the request. If
+          # the server doesn't support this method, it returns `UNIMPLEMENTED`.
+          #
+          # NOTE: the `name` binding below allows API services to override the
+          # binding to use different resource name schemes, such as
+          # `users/*/operations`.
+          #
+          # @param name [String]
+          #   The name of the operation collection.
+          # @param filter [String]
+          #   The standard list filter.
+          # @param page_size [Integer]
+          #   The maximum number of resources contained in the underlying API
+          #   response. If page streaming is performed per-resource, this
+          #   parameter does not affect the return value. If page streaming is
+          #   performed per-page, this determines the maximum number of
+          #   resources in a page.
+          # @param options [Google::Gax::CallOptions]
+          #   Overrides the default settings for this call, e.g, timeout,
+          #   retries, etc.
+          # @return [Google::Gax::PagedEnumerable<Google::Gax::Operation>]
+          #   An enumerable of Google::Gax::Operation instances. See
+          #   Google::Gax::PagedEnumerable documentation for other operations
+          #   such as per-page iteration or access to the response object.
+          # @raise [Google::Gax::GaxError] if the RPC is aborted.
+          # @example
+          #   require "google/cloud/speech"
+          #
+          #   speech_client = Google::Cloud::Speech.new version: :v1
+          #
+          #   # List all callbacks.
+          #   speech_client.list_operations "*", "" do |op|
+          #     # Process error operations.
+          #     log_error op.error if op.error?
+          #
+          #     if op.done?
+          #       # Process completed operations.
+          #       log_finished op.response, op.metadata
+          #     else
+          #       # Process pending operations.
+          #       log_pending op.name, op.metadata
+          #     end
+          #   end
+          #
+          def list_operations name, filter, page_size: nil, options: nil
+            paged_enum = @operations_client.list_operations \
+              name, filter, page_size: page_size, options: options
+            paged_enum.lazy.map do |proto_op|
+              wrap_proto_op_with_gax_op proto_op, options
+            end
+          end
+
+          protected
+
+          ##
+          # @private
+          # Lookup the Protobuf class from the type URL
+          def wrap_proto_op_with_gax_op proto_op, options
+            Google::Gax::Operation.new(
+              proto_op,
+              @operations_client,
+              get_type_class_from_proto_any(proto_op.response),
```

Comment:
After sleeping on it, I don't think this approach will work. Inspecting the `Google::Longrunning::Operation` for the response type before creating the `Google::Gax::Operation` may give the wrong response type. This is because if the `Google::Gax::Operation` is not complete, it can be refreshed and the underlying proto object changed. Only after refresh will the response type be correct.

I think we should go back to what the approach used in the prior hand-written version of this library. We can hard-code these types because there is only one kind of operation on this API. Thoughts?

https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/google-cloud-speech/v0.29.0/google-cloud-speech/lib/google/cloud/speech/service.rb#L91-L94

Reply:
How about waiting until it's accessed. Isn't that what you were trying to do with the GAX change?

Reply:
I have updated this PR to use the same types as the old version and the same types as the GAPIC client. This should work for Speech but is not a generalized solution. We still need a change like googleapis/gax-ruby#134 for a generalized solution.